tests/components/hikvision/test_init.py

```diff
@@ -60,3 +70,34 @@ async def test_setup_entry_no_device_id(
     await hass.async_block_till_done()
 
     assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+@pytest.mark.parametrize("ssl", [False, True])
+async def test_setup_entry_passes_ssl_parameter(
+    hass: HomeAssistant,
+    mock_hikcamera: MagicMock,
+    ssl: bool,
+) -> None:
+    """Test that ssl parameter is passed to HikCamera."""
+    mock_config_entry = MockConfigEntry(
+        title="Test Camera",
+        domain=DOMAIN,
+        data={
+            CONF_HOST: TEST_HOST,
+            CONF_PORT: TEST_PORT,
+            CONF_USERNAME: TEST_USERNAME,
+            CONF_PASSWORD: TEST_PASSWORD,
+            CONF_SSL: ssl,
+        },
+        unique_id="test_device_id",
+    )
```

Comment:
Use your fixture, you can update it just for this test using hass.config_entries.async_update_entry as you probably don't need this for every test